noir/noir-repo/compiler/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -775,6 +783,23 @@ impl<'block> BrilligBlock<'block> {
         self.brillig_context.set_allocated_registers(registers);
     }
 
+    fn needs_runtime_array_index_check(
+        &self,
+        array: ValueId,
+        index: ValueId,
+        dfg: &DataFlowGraph,
+    ) -> bool {
+        let array_type = dfg.type_of_value(array);
+        let Type::Array(item_types, len) = array_type else {
+            return true;
+        };
+        let Some(index) = dfg.get_numeric_constant(index) else {
+            return true;
+        };
+        // We skip the runtime check if the index is a constant and the index is within bounds
+        index >= (len * item_types.len()).into()
+    }
+
```

Comment:
We have `dfg.is_safe_index` which this should be indentical to.

Reply:
Didn't know that!